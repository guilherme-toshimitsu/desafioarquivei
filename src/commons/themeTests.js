import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
