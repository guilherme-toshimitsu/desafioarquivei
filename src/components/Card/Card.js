import { memo } from "react";
import styled from "styled-components";
import { breakpoints, points } from "@commons/theme";

const StyledCard = styled.div`
  overflow: hidden;
  padding: 24px;
  margin: 48px auto 0;
  background-color: #ffffff;
  box-shadow: 0 0 0.2em 0.2em #a0a0a0;
  border-radius: 4px;
  max-width: 420px;
  min-height: 400px;

  ${breakpoints.sm} {
    margin:  0 auto;
    padding: 8px,
    max-width: 320px
  }

`;

export default memo(StyledCard);
