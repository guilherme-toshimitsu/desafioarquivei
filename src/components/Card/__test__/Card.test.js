import React from "react";
import Card from "..";
import { renderWithTheme } from "@commons/themeTests";

describe("<Card />", () => {
  it("creates a Button component", () => {
    const wrapper = renderWithTheme(<Card />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
