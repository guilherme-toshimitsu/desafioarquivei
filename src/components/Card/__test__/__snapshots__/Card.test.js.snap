// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> creates a Button component 1`] = `
<DocumentFragment>
  <div
    class="Card__StyledCard-sc-19fims7-0 HtJiy"
  />
</DocumentFragment>
`;
